### PR TITLE
fix: reworked HTTP status code handling for 2xx and 3xx ranges (MWTELE-68, MWTELE-69)

### PR DESCRIPTION
**Pull request by jponge**
This ensures any 2xx response is considered valid. The client now follows redirects (3xx range).

JIRA: https://issues.redhat.com/browse/MWTELE-68
JIRA: https://issues.redhat.com/browse/MWTELE-69

### CODE REVIEW

File: runtime/src/main/java/com/redhat/insights/core/httpclient/InsightsJdkHttpClient.java [hunk 1]
```diff
@@ -138,35 +138,40 @@ protected void sendInsightsReportWithClient(
...
+                          ERROR_HTTP_SEND_AUTH_ERROR, "Authentication missing from request");
+                    case 413:
+                      throw new InsightsException(ERROR_HTTP_SEND_PAYLOAD, "Payload too large");
+                    case 415:
+                    default:
+                      throw new InsightsException(
+                          ERROR_HTTP_SEND_INVALID_CONTENT_TYPE,
+                          "Content type of payload is unsupported");
+                  }
+                case 5:
```
Comment (jponge) [inline runtime/src/main/java/com/redhat/insights/core/httpclient/InsightsJdkHttpClient.java:170]: Note: I have deliberately kept multiple `case` label statements based on the original OpenAPI spec of the Go insights backend.

**Comment by mocenas** [comment]
LGTM

File: runtime/src/main/java/com/redhat/insights/core/httpclient/InsightsJdkHttpClient.java (hunk 1 above)
Comment (ehsavoie) [reply runtime/src/main/java/com/redhat/insights/core/httpclient/InsightsJdkHttpClient.java:170]: Why not defining an enum for each 'category': could be more readable WDYT ?

Comment (ehsavoie) [reply runtime/src/main/java/com/redhat/insights/core/httpclient/InsightsJdkHttpClient.java:170]: Shouldn't we deal with 403 as maybe the subscription is no longer active or something like this ?

**Pull request merged**